shared/src/components/activation/ActivationClickTarget.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import { Subscription } from 'rxjs'
+import { ActivationStatus } from './Activation'
+
+export interface ActivationClickTargetState {
+    activated?: boolean
+    animate?: boolean
+}
+
+export interface ActivationClickTargetProps {
+    /**
+     * The activation status to update when the element is clicked.
+     */
+    activation?: ActivationStatus
+
+    /**
+     * The activation step keys to set to true when the element is clicked.
+     */
+    activationKeys: string[]
+
+    /**
+     * If defined, the component will absorb the click (preventing propagation), pause
+     * (enough time for the activation animation to finish), and then invoke the
+     * specified function after the pause.
+     */
+    pauseAndRetrigger?: () => void
+}
+
+/**
+ * Wraps another component where a user click should trigger the completion of one or more
+ * activation steps. A user click that results in one or more incomplete steps to be
+ * completed will trigger the activation animation on this component. Clicks that
+ * do not trigger incomplete steps to complete will not trigger any animation.
+ */
+export class ActivationClickTarget extends React.PureComponent<ActivationClickTargetProps, ActivationClickTargetState> {
+    private subscriptions = new Subscription()
+    constructor(props: ActivationClickTargetProps) {
+        super(props)
+        this.state = {}
+    }
+    public componentDidMount(): void {
+        if (this.props.activation) {
+            this.subscriptions.add(
+                this.props.activation.completed.subscribe(completed => {
+                    if (!completed) {
+                        return
+                    }
+                    // Completed list is available, so now determine whether it's activated by looking at what's completed
+                    let activated = true
```

Comment:
```suggestion
                    const activated = this.props.activationKeys.every(k => completed[k] !== undefined)
```

removes the need for the `for` loop below